File: bfs.py
class vertex(object):
    def __init__(self, name):
        self.name=name
        self.colour=None
        self.d=None
        self.p=None

s=vertex('s')
b=vertex('b')
c=vertex('c')
d=vertex('d')
e=vertex('e')
f=vertex('f')
g=vertex('g')

V=[s,b,c,d,e,f,g]

for i in V:
    i.colour="white"
    i.d = None
    i.p=None

V[0].colour = "gray"
V[0].d=0
V[0].p=None

G={s:[b,c], b:[s,d,g], c:[s,f], d:[b,e], e:[d], f:[c], g:[b]}

Q=[]
Q.append(V[0])
while(len(Q)!=0):
    u=Q.pop(0)
    for v in G[u]:
        if v.colour=="white":
            v.colour="gray"
            v.d=u.d+1
            v.p=u
            Q.append(v)
            print(v.name)        
    u.colour="black"    

File: bfs.cpp
#include <iostream>
#include <queue>
#include <string>
#include <bits/stdc++.h>
#include <list>
#include <vector>
using namespace std;


class vertex{
	string name;
	string colour;
public:
	vertex(){};
	vertex(string s){name=s; colour="white";}	
	string getName(){return name;}
	string getColour(){return colour;}
	void setColour(string c){colour = c;}
};

 
int main()
{
	int n;
	cout << "Enter the number of vertices: ";
	cin >> n;
    vector <vertex> vertices;
    cout << "Enter the name of vertices" << endl;
    for(int i=0; i<n; i++)
    {
    	string name;
    	cin >> name;
    	vertices.push_back(vertex(name));
    }
    

    list <vertex *> adjlist[n];
    unordered_map<string, list<vertex*>> Adj;

    for(int i=0; i<n; i++)
    {
    	cout << "Enter the number of vertices adjacent to " << vertices[i].getName() << ": " << endl;
    	int numOfAdjV;
    	cin >> numOfAdjV;
    	for(int j=0; j<numOfAdjV; j++)
    	{
    		cout << "Enter the name of adjacent vertex "<< j+1 << "/" << numOfAdjV << " : ";
    		string nameOfAdjV;
    		cin >> nameOfAdjV;

    		for(int k=0; k<n; k++)
    			if(nameOfAdjV == vertices[k].getName())
    				adjlist[i].push_back(&vertices[k]);
    	}

    	Adj[vertices[i].getName()] = adjlist[i];
    }


    queue<vertex *> Q;
    vertices[0].setColour("black");
    Q.push(&vertices[0]);


    while(!Q.empty())
    {
    	vertex *v = Q.front();
    	Q.pop();

    	for(list <vertex *> :: iterator i=Adj[v->getName()].begin(); i!=Adj[v->getName()].end(); i++)
    	{
    		if((*i)->getColour() == "white")
    		{
    			(*i)->setColour("gray");
    			Q.push(*i);    
    			cout << (*i)->getName() << endl;			
    		}
    	}
    	
    	v->setColour("black");    	

    }
}